docs/managing-fork.md

```diff
@@ -0,0 +1,152 @@
+  origin    ssh://git@github.com/elastisys/kubespray (push)
+  upstream  ssh://git@github.com/kubernetes-sigs/kubespray.git (fetch)
+  upstream  ssh://git@github.com/kubernetes-sigs/kubespray.git (push)
+
+# If not:
+$ git remote add upstream ssh://git@github.com/kubernetes-sigs/kubespray.git
+```
+
+### Step 2
+
+Make sure the master branch is up to date with the latest upstream release version:
+
+```console
+$ cd kubespray
+$ git fetch --all
+$ git switch master
+
+$ git status # Ensure these lines match
+On branch master
+Your branch is up to date with 'origin/master'.
+
+$ git branch master -u origin/master # Else if your local master branch tracks another remote branch, rerun git status.
+$ git reset --hard origin/master # Else if your local master branch has diverged from origin/master.
+$ git rebase -i <upstream-version>
+# Keep relevant commits we've previously included and resolve all merge conflicts.
+$ git log
+# Ensure the commit history has not diverged from upstream/master.
+# The first commit after ours should be marked "upstream/master".
+$ git status # Ensure these lines match
+  On branch master
+  Your branch and 'origin/master' have diverged,
+  and have N and M different commits each, respectively.
+$ git push --force-with-lease
+```
+
+### Step 3
+
+Create a new feature branch and start developing:
+
+```sh
+git switch -c <development-branch>
+```
+
+Create a pull request to master within our fork, get it reviewed, and then merged.
+
+### Step 4
+
+Create or reuse a release branch for the fork based on an upstream release.
+
+The releases will follow the version of the upstream version, with the version string `<version>-ck8s`.
+
+- `<version>` - denotes the current version of the upstream version.
+- `ck8s` - added to prevent confusion with upstream release branch
+
+```sh
+git fetch --all
+
+# if a release/patch exists and you only want to add a fix to the fork:
+git switch "release-X.Y.Z-ck8s"
+
+# if no release branch exists in the fork:
+git switch X.Y.Z # upstream release
+git switch -c "release-X.Y.Z-ck8s"
+
+git push -u origin "release-X.Y.Z-ck8s"
+```
+
+### Step 5
+
+Backport the fixes to the release branch.
+
+```sh
+git cherry-pick COMMIT-SHA [COMMIT-SHA ...]
+git push
+```
+
+### Step 6
+
+Update the submodule reference in compliantkubernetes-kubespray.
+
+```sh
+cd compliantkubernetes-kubespray
+git switch master
+git pull
+git submodule update --init
+git switch -c <development-branch>
+
+cd kubespray
+git switch "release-X.Y.Z-ck8s"
+cd ..
+git add kubespray
+git commit -m "Upgrade kubespray fork ..."
+git push
+# open pull request towards main branch of compliantkubernetes-kubespray
+```
+
+### Step 7
+
+Tag the latest commit in the fork release branch. `patch-number` starts at 1 and increments by one for every new `compliantkubernetes-kubespray` release.
+
+```sh
+cd <fork-directory>
+git switch "release-X.Y.Z-ck8s"
+git pull
+git tag -a "vX.Y.Z-ck8s<patch-number>"
+git push --tags
+```
```

Comment:
Isn't it better if we do this during the release step?

Reply:
Yes, it's included in the release steps, I just kept it here to sort of give the whole picture. Added a disclaimer now, or should I remove it from here entirely?